app/decorators/odk/qing_group_decorator.rb

```diff
@@ -50,6 +50,8 @@ def header_bind_tag(xpath_prefix: "/data")
     def body_tags(xpath_prefix:, **_options)
       return main_body_tags(xpath_prefix) if root?
 
+      return safe_str unless visible? && enabled?
```

Comment:
This guard is new since pairing, to prevent rendering hidden qing groups in the ODK body. I don't quite understand:
1. why this guard wasn't previously necessary with disabled qing groups (though the guard was used before for regular qings, see qing_decorator above)
2. why `visible?` isn't defined here (causes spec failures) even though both qing and qing_group inherit from form_item where `visible?` is defined

Can you please help me understand? Happy to pair if it's easier.